src/api/app/models/attrib_type.rb

```diff
@@ -9,68 +16,64 @@ class AttribType < ActiveRecord::Base
   has_many :allowed_values, :class_name => 'AttribAllowedValue', dependent: :delete_all
   has_many :attrib_type_modifiable_bies, :class_name => 'AttribTypeModifiableBy', dependent: :delete_all
 
+  #### Callbacks macros: before_save, after_save, etc.
+  #### Scopes (first the default_scope macro if is used)
+  #### Validations macros
   validates :name, presence: true
 
-  class << self
-    class UnknownAttributeTypeError < APIException
-      setup 'unknown_attribute_type', 404, 'Unknown Attribute Type'
-    end
+  #### Class methods using self. (public and then private)
+  def self.find_by_name!(name)
+     find_by_name(name, true)
+  end
 
-    def find_by_name!(name)
-       find_by_name(name, true)
+  def self.find_by_name(name, or_fail=false)
+    name_parts = name.split(/:/)
+    if name_parts.length != 2
+      raise ArgumentError, "attribute '#{name}' must be in the $NAMESPACE:$NAME style"
     end
+    find_by_namespace_and_name(name_parts[0], name_parts[1], or_fail)
+  end
 
-    def find_by_name(name, or_fail=false)
-      name_parts = name.split(/:/)
-      if name_parts.length != 2
-        raise ArgumentError, "attribute '#{name}' must be in the $NAMESPACE:$NAME style"
-      end
-      find_by_namespace_and_name(name_parts[0], name_parts[1], or_fail)
-    end
+  def self.find_by_namespace_and_name!(namespace, name)
+     find_by_namespace_and_name(namespace, name, true)
+  end
 
-    def find_by_namespace_and_name!(namespace, name)
-       find_by_namespace_and_name(namespace, name, true)
+  def self.find_by_namespace_and_name(namespace, name, or_fail=false)
+    unless namespace and name
+      raise ArgumentError, "Need namespace and name as parameters"
     end
-
-    def find_by_namespace_and_name(namespace, name, or_fail=false)
-      unless namespace and name
-        raise ArgumentError, "Need namespace and name as parameters"
-      end
-      ats = joins(:attrib_namespace).where("attrib_namespaces.name = ? and attrib_types.name = ?", namespace, name)
-      if or_fail and ats.count != 1
-        raise UnknownAttributeTypeError, "Attribute Type #{namespace}:#{name} does not exist"
-      end
-      ats.first
+    ats = joins(:attrib_namespace).where("attrib_namespaces.name = ? and attrib_types.name = ?", namespace, name)
+    if or_fail and ats.count != 1
+      raise UnknownAttributeTypeError, "Attribute Type #{namespace}:#{name} does not exist"
     end
+    ats.first
   end
 
+  #### To define class methods as private use private_class_method
+  #### private
+  #### Instance methods (public and then protected/private)
   def namespace
-    self.attrib_namespace.name
+    attrib_namespace.name
   end
 
   def fullname
-    return "#{self.attrib_namespace}:#{self.name}"
+    "#{attrib_namespace}:#{name}"
   end
 
-  def create_one_rule(m)
-    if m["user"].blank? and m["group"].blank? and m["role"].blank?
+  def create_one_rule(node)
```

Comment:
Isn't that the same function in `` src/api/app/models/attrib_namespace.rb ``?
Can we DRY that?